issues.py

```diff
@@ -36,53 +48,159 @@ def get_issue_dict(alert, project):
+            GITHUB_URL + "/" + str(issue_id) + "/" + "labels",
+            data=json.dumps([SUPPRESSION_LABEL]),
+        )
+        if r.ok:
+            return jsonify({"issue-id": issue_id}), 200
+        if r.status_code == 404 and r.json()["message"] == "Not Found":
+            # if we were trying to suppress, we don't worry about not finding it
+            return jsonify({"issue-id": issue_id}), 200
+        else:
+            return jsonify({"error": 500}), 500
+
+    if transition == "unsuppress":
+
+        r = sess_github.delete(
+            "/".join([GITHUB_URL, str(issue_id), "labels", SUPPRESSION_LABEL])
+        )
+
+        # if the label was not present on the issue, we don't let this worry us
+        if not r.ok and r.json().get("message") == "Label does not exist":
+            r.status_code = 200
+
+        if r.ok:  # we ensure that the ticket is open in the issue tracker
+            r = sess_github.patch(
+                GITHUB_URL + "/" + str(issue_id), data=json.dumps({"state": "open"})
+            )
+
+        if r.ok:
+            return jsonify({"issue-id": issue_id}), 200
+        if r.status_code == 404 and r.json()["message"] == "Not Found":
+            # using 410 we indicate to LGTM that the issue needs to be recreated
+            return jsonify({"error": "gone"}), 410
+        else:
+            return jsonify({"error": 500}), 500
+
+    # when the transition is not recognised, we return a bad request response
+    return (jsonify({"message": "unknown transition type - %s" % transition}), 400)
 
-    if not r.ok:
-        return r.content, r.status_code
 
-    return jsonify({"issue-id": issue_id}), r.status_code
+@app.route("/github", methods=["POST"])
+def github_webhook():
+    """Handle POST requests coming from GitHub, and pass a translated request to LGTM"""
+
+    if not auth_is_valid(
+        request.headers.get("X-Hub-Signature", "not-provided").split("=")[-1]
+    ):
+        return jsonify({"message": "Unauthorized"}), 403
+
+    json_dict = request.get_json()
+
+    action = json_dict.get("action")
+
+    if action not in ["labeled", "unlabeled"]:
+        return jsonify({"status": 200}), 200
+
+    label = json_dict.get("label", {})
+
+    if label.get("name", "") != SUPPRESSION_LABEL:
+        return jsonify({"status": 200}), 200
+
+    issue_id = str(json_dict["issue"]["number"])
+
+    # When we were responsible for changing the tag, we don't want to pass the webhook back again.
+    if json_dict["sender"]["login"] == GITHUB_BOT_USERNAME:
+        return jsonify({"status": 200}), 200
+
+    translator = {"labeled": "suppress", "unlabeled": "unsuppress"}
+
+    payload = json.dumps({"issue-id": issue_id, "transition": translator[action]})
+
+    headers = {
+        "X-LGTM-Signature": hmac.new(KEY, payload.encode("utf-8"), "sha1").hexdigest()
+    }
+
+    r = sess_lgtm.post(LGTM_URL, data=payload, headers=headers)
+
+    if r.ok:
+        return jsonify({"status": 200}), 200
+    else:
```

Comment:
Returning an error code here would cause GitHub to re-send the webhook again later right? Is that intended?

Reply:
I found no documentation about GitHub retrying failed webhooks, and when testing it never seems to happen. What it does do is mark the request as failed when viewing the webhook config on GitHub, and allows one to see what the error message was.

Reply:
Would it be legitimate in this case to just pass on the exact response from LGTM, for maximum debugging value?

Reply:
I'm fine with that.